internal/locate/region_request.go

```diff
@@ -245,6 +245,8 @@ type replica struct {
 	peer     *metapb.Peer
 	epoch    uint32
 	attempts int
+	// deadlineExceededErr indicates the replica is already tried, but the received deadline exceeded error.
+	deadlineExceededErr bool
```

Comment:
It means `deadlineExceededErr` encountered using `ConfigurableTimeout` right, maybe `deadlineErrUsingConfTimeout` is better?